### PR TITLE
perf: add jobs applications data in database

### PR DESCRIPTION
**Pull request by cp-mansi-d**
## Review checklist

**Note:** Make sure all points should be checked before submiting PR for review, otherwise PR will be cosidered invalid.

- [x] Follow guidelines for creating branch name, adding commit messages or PR title
- [x] Add MR description if applicable
- [x] Follow guidelines for naming conventions everywhere (i.e files/folders, data structures)
- [x] Reuse code (if the same code is written twice, make it common and reuse it at both places)
- [x] Remove unused or commented code if not required
- [x] Make MR self-approved - review MR as a reviewer and give it self-approval if everything is ok, if not make the required changes
- [x] In progress MR should be marked as draft
- [x] Make MR mark as ready before submitting it for review
- [ ] Add images, videos, or gifs of UI 
- [ ] Add page speed score screenshot of UI (It should be > 85 for mobile and > 90 for desktops)

**Design should be tested on:**
- [ ] Inspector (all iPhones and android devices like samsung, motorola)
- [ ] Desktop (chrome, firefox, safari)
- [ ] Laptop / Macbook  (chrome, firefox, safari)
- [ ] iPhone(safari)
- [ ] Available android devices  (chrome, firefox)
- [ ] iPad(safari)
- [ ] Tablet (chrome, firefox)

**Description**
- added new feature - When user sends job applications then added data to database and showed inside website admin panel instead of sending email.
- and also added that when a user uploads their resume, that resume is automatically uploaded to the aws bucket and that resume url is stored inside the job application table. So hr will see that resume properly.

### CODE REVIEW

File: jobs/jobs.go [hunk 1]
```diff
@@ -163,26 +166,101 @@ func (repository *CareerRepository) SendCareerMail(c *gin.Context) {
...
+	}
 
-	statusCode := repository.UtilsRepo.SendEmail(nil, emailTemplate)
+	return nil
+}
 
-	if statusCode != 0 {
-		c.AbortWithStatus(statusCode)
-		return
+func uploadResumeToS3(fileHeader *multipart.FileHeader) (string, error) {
```
Comment (cp-sumi-k) [inline jobs/jobs.go:222]: Let's move this method to utils/s3.go

File: jobs/jobs.go [hunk 2]
```diff
@@ -163,26 +166,101 @@ func (repository *CareerRepository) SendCareerMail(c *gin.Context) {
...
+		return "", err
+	}
+
+	// Generate the URL for the uploaded resume
+	resumeURL := fmt.Sprintf("https://%s.s3.%s.amazonaws.com/resumes/%s", "canopas-website", os.Getenv("REGION"), fileName)
+
+	return resumeURL, nil
+}
+
+func generateUniqueFileName(filename string) string {
```
Comment (cp-sumi-k) [inline jobs/jobs.go:259]: Move this method to utils/helpers.go

File: vue-frontend/src/pages/jobs/apply/[id].vue [hunk 3]
```diff
@@ -536,12 +535,14 @@ export default {
             : "NA"
         );
 
+        formData.append("resume", this.file, fileName);
+        formData.append("position", this.job.title);
```
Comment (cp-sumi-k) [inline vue-frontend/src/pages/jobs/apply/[id].vue:539]: Any need for this as we are setting it as `job_title`?

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

File: vue-frontend/src/pages/jobs/apply/[id].vue (hunk 3 above)
Comment (cp-mansi-d) [reply vue-frontend/src/pages/jobs/apply/[id].vue:539]: no need extra, I thought that when we remove the email sending functionality, the other member will easily understand that we have used this job title value for the position.

**Review by cp-jagruti-a — APPROVED** [review]

File: jobs/jobs.go [hunk 4]
```diff
@@ -163,6 +162,37 @@ func (repository *CareerRepository) SendCareerMail(c *gin.Context) {
 		go repository.UtilsRepo.SaveJobsToSpreadSheet(records)
 	}
 
+	repository.insertJobApplication(input)
```
Comment (cp-sumi-k) [inline jobs/jobs.go:165]: Let's check error here as this method returning it

File: jobs/jobs.go [hunk 5]
```diff
@@ -163,6 +162,37 @@ func (repository *CareerRepository) SendCareerMail(c *gin.Context) {
...
+
+	repository.sendEmailApplication(c, input)
```
Comment (cp-sumi-k) [inline jobs/jobs.go:167]: Can be renamed to `SendJobsApplicationMail`

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

**Review by cp-sumi-k — APPROVED** [review]

**Review by cp-sumi-k — APPROVED** [review]

**Pull request merged**